kmapper/kmapper.py

```diff
@@ -522,6 +530,36 @@ def map(
             self._summary(graph, str(datetime.now() - start))
 
         return graph
+    
+    def _remove_duplicate_nodes(self, nodes):
```

Comment:
It'd be really nice if this function had no side effects, i.e., returns a different nodes object.

Reply:
My apologies, made a bad assumption about python value / reference passing. Will fix.

Reply:
pushed a fix that uses shallow `.copy` which is good enough since only keys are being dropped -- values not edited